noir-projects/aztec-nr/aztec/src/keys/ephemeral.nr

```diff
@@ -0,0 +1,32 @@
+use std::{
+    embedded_curve_ops::fixed_base_scalar_mul, field::bn254::decompose,
+    hash::from_field_unsafe as fr_to_fq_unsafe,
+};
+
+use dep::protocol_types::{point::Point, scalar::Scalar};
+
+use crate::oracle::random::random;
+
+/// Converts a base field element to scalar field element.
+/// This is fine because modulus of the base field is smaller than the modulus of the scalar field.
+fn fr_to_fq(r: Field) -> Scalar {
```

Comment:
```suggestion
/// Converts a base field element to scalar field element.
/// This is lossless because the modulus of the base field is smaller than the modulus of the scalar field.
fn fr_to_fq(r: Field) -> Scalar {
```